src/test/java/wolox/training/controllers/UserControllerTest.java

```diff
@@ -0,0 +1,166 @@
+                .andExpect(content().json("{\"id\":null,\"username\":\"marti\",\"name\":\"Martina\",\"birthdate\":\"1993-12-09\",\"books\":[{\"id\":null,\"genre\":\"novela grafica\",\"author\":\"DC: Alan Moore\",\"image\":\"image.png\",\"title\":\"Batman: The Killing Joke\",\"subtitle\":\"A subtitle\",\"publisher\":\"Publisher\",\"year\":\"1988\",\"pages\":100,\"isbn\":\"ABCD1234\"}]}"));
+    }
+
+    @Test
+    public void whenFindAUserByUsername_thenReturnTheUser() throws Exception {
+        Mockito.when(mockUserRepository.findByUsername("Username")).thenReturn(userTest);
+        mvc.perform(get("/api/users/Username").contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(content().json("{\"id\":null,\"username\":\"marti\",\"name\":\"Martina\",\"birthdate\":\"1993-12-09\",\"books\":[{\"id\":null,\"genre\":\"novela grafica\",\"author\":\"DC: Alan Moore\",\"image\":\"image.png\",\"title\":\"Batman: The Killing Joke\",\"subtitle\":\"A subtitle\",\"publisher\":\"Publisher\",\"year\":\"1988\",\"pages\":100,\"isbn\":\"ABCD1234\"}]}"));
+    }
+
+    @Test
+    public void whenUpdateAUser_thenTheUserIsUpdated() throws Exception {
+        userTest.setName("Other name");
+        userTest.setId(1L);
+        Mockito.when(mockUserRepository.findById(1L)).thenReturn(ofNullable(userTest));
+        mvc.perform(put("/api/users/1").contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsString(userTest)))
+                .andExpect(status().isOk());
+    }
+
+    @Test
+    public void whenDeleteAUser_thenTheUserIsDeleted() throws Exception {
+        Mockito.when(mockUserRepository.findById(1L)).thenReturn(ofNullable(userTest));
+        mvc.perform(delete("/api/users/1").contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk());
+    }
+
+    @Test
+    public void whenDeleteAUserDontExists_thenReturnNotFound() throws Exception {
+        Mockito.when(mockUserRepository.findById(1L)).thenReturn(ofNullable(userTest));
+        mvc.perform(delete("/api/users/2").contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isNotFound());
+    }
+
+    @Test
+    public void whenAddABookToUsers_thenIsPersistedInTheUserList() throws Exception {
+        Mockito.when(mockBookRepository.findById(1L)).thenReturn(ofNullable(bookTest));
+        Mockito.when(mockUserRepository.findById(1L)).thenReturn(ofNullable(userTest));
+        mvc.perform(put("/api/users/1/books").contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsString(bookTest)))
+                .andExpect(status().isCreated());
+    }
+
+    @Test
+    public void whenAddABookToAUsersThatNotExists_thenReturnNotFound() throws Exception {
+        Mockito.when(mockBookRepository.findById(1L)).thenReturn(ofNullable(bookTest));
+        Mockito.when(mockUserRepository.findById(1L)).thenReturn(ofNullable(userTest));
```

Comment:
Here you should mock the exception throwing
```suggestion
        Mockito.when(mockUserRepository.findById(1L)).thenThrow(UserNotFoundException.class);
```